### PR TITLE
add a delegate swap check

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Put the preSwapDelegate check within the preSwapChecker
I separated the delegate tests into their own file as well (more forward looking for if we create a preSwapDelegateWrapper check function as well, i'll likely separate the smart contracts as well).

Quick description:
- [X] New features

## Changes

Added a prewap checker when using the delegate

## Tests

  46 passing (9s)

## Test Coverage

100%

### CODE REVIEW

File: utils/pre-swap-checker/contracts/PreSwapChecker.sol [hunk 1]
```diff
@@ -37,6 +38,77 @@ contract PreSwapChecker {
...
+
+    if (swapErrorCount > 0) {
+      errorCount = swapErrorCount;
+      // copies over errors from checkSwapSwap to be outputted
+      for (uint256 i = 0; i < swapErrorCount; i++) {
+        errors[i] = swapErrors[i];
+      }
+    }
+
+    // signature must be filled in order to use the Wrapper
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/contracts/PreSwapChecker.sol]: `Wrapper` -> `Delegate`

File: utils/pre-swap-checker/contracts/PreSwapChecker.sol [hunk 2]
```diff
@@ -10,6 +10,7 @@ import "@airswap/swap/contracts/Swap.sol";
 import "@airswap/transfers/contracts/TransferHandlerRegistry.sol";
 import "@airswap/tokens/contracts/interfaces/IWETH.sol";
 import "@airswap/wrapper/contracts/Wrapper.sol";
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/contracts/PreSwapChecker.sol]: What is Wrapper.sol imported for? Can't find it anywhere in the contract

Comment (alicevhenshaw) [reply utils/pre-swap-checker/contracts/PreSwapChecker.sol]: Also where the contract uses `Swap(...)` can it not just use `ISwap(...)`? e.g.
`TransferHandlerRegistry tokenRegistry = Swap(swap).registry();`

Comment (poiresel) [reply utils/pre-swap-checker/contracts/PreSwapChecker.sol]: use of registry was never put into ISwap, only in Swap. maybe want to remedy that now.

Comment (poiresel) [reply utils/pre-swap-checker/contracts/PreSwapChecker.sol]: actually unable to do that as TransferHandlerRegistry is a contract not an interface

Comment (poiresel) [reply utils/pre-swap-checker/contracts/PreSwapChecker.sol]: i can be consistent and just use Swap everything and not import both Swap and ISwap though.

File: utils/pre-swap-checker/test/PreDelegateChecker.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+          [tokenAST, 0],
+          [tokenDAI, 1000],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+  })
+
+  describe('Approving...', async () => {
+    it('Checks approvals (Alice 250 AST and 0 DAI, Bob 0 AST and 500 DAI)', async () => {
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/test/PreDelegateChecker.js]: Its 400 and 1000 not 250 and 500

File: utils/pre-swap-checker/test/PreDelegateChecker.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+        bobAddress,
+        swapAddress,
+        GANACHE_PROVIDER
+      )
+
+      // // create a rule for AST/DAI for alice
+      // await aliceDelegate.setRule(
+      //   tokenAST.address,
+      //   tokenDAI.address,
+      //   1000,
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/test/PreDelegateChecker.js]: Delete the commented lines

File: utils/pre-swap-checker/contracts/PreSwapChecker.sol (hunk 2 above)
Comment (poiresel) [reply utils/pre-swap-checker/contracts/PreSwapChecker.sol]: never mind, ive resolved.

File: utils/pre-swap-checker/contracts/Imports.sol [hunk 5]
```diff
@@ -8,5 +8,7 @@ import "@airswap/transfers/contracts/TransferHandlerRegistry.sol";
 import "@airswap/transfers/contracts/handlers/ERC20TransferHandler.sol";
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/contracts/Imports.sol]: Can remove `import "@airswap/transfers/contracts/TransferHandlerRegistry.sol";` from Imports.sol - it's imported in the PreSwapChecker already.

**Pull request merged**